### PR TITLE
feat(ada): migration from cardano-rest to cardano-graphql

### PR DESCRIPTION
**Pull request by marianogoldman**
## Checklist

- [ ] Documentation _(if necessary)_
- [ ] Tests _(if necessary)_
- [ ] Ready to be merged

### CODE REVIEW

File: packages/platform-sdk-ada/src/dto/transaction.ts [hunk 1]
```diff
@@ -69,10 +94,12 @@ export class TransactionData extends DTO.AbstractTransactionData implements Cont
 	}
 
 	public isSent(): boolean {
+		// FIXME Need to find a way to determine this
 		return this.data.direction === "outgoing";
 	}
 
 	public isReceived(): boolean {
+		// FIXME Need to find a way to determine this
```
Comment (faustbrian) [inline packages/platform-sdk-ada/src/dto/transaction.ts:102]: We could probably check if the outputs contain one of our addresses.

File: packages/platform-sdk-ada/src/dto/transaction.ts [hunk 2]
```diff
@@ -69,10 +94,12 @@ export class TransactionData extends DTO.AbstractTransactionData implements Cont
 	}
 
 	public isSent(): boolean {
+		// FIXME Need to find a way to determine this
```
Comment (faustbrian) [inline packages/platform-sdk-ada/src/dto/transaction.ts:97]: We could probably check if the inputs contain one of our addresses.

File: packages/platform-sdk-ada/src/dto/transaction.ts [hunk 3]
```diff
@@ -46,18 +51,29 @@ export class TransactionData extends DTO.AbstractTransactionData implements Cont
...
 			(output: Contracts.KeyValuePair) =>
 				new DTO.UnspentTransactionData({
-					amount: BigNumber.make(output.amount.quantity),
+					amount: BigNumber.make(output.value),
 					addresses: [output.address],
 				}),
 		);
 	}
 
 	public amount(): BigNumber {
```
Comment (faustbrian) [inline packages/platform-sdk-ada/src/dto/transaction.ts:60]: Could we use the `totalOutput` property for this? It seems to exclude the `fee` but I would need to check a transaction that contained changed to see if that is also taken into account.

**Comment by faustbrian** [comment]
Merging this to work on some other things. Remaining changes can be submitted through another PR.

File: packages/platform-sdk-ada/src/dto/transaction.ts (hunk 3 above)
Comment (marianogoldman) [reply packages/platform-sdk-ada/src/dto/transaction.ts:60]: @faustbrian Change is also part of the output. That's why I had to do that ridiculous assumption (that change will be the last output address) and the the transferred amount are all the others.

Comment (faustbrian) [reply packages/platform-sdk-ada/src/dto/transaction.ts:60]: Will have a look at some other possible solutions then.

**Pull request merged**